### DISCUSSION TITLE
What lock will be acquired when creating new partition

### DISCUSSION THREAD

**Discussion by jhchee** [pgpartman/pg_partman#687] [Q&A] [answered]
Hi community, I have a question regarding run_maintenance_proc in particular how it creates partition. 

Will the create partition operation obtain an  `ACCESS EXCLUSIVE` lock, or it is run in a manner which it first creates the partition table and later attaching to the parent table (this obtains `SHARE UPDATE EXCLUSIVE` lock)

Reference: https://maksimrv.medium.com/a-one-story-about-problem-with-partitioned-table-in-postgresql-a089916fd164

**Comment by keithf4** [comment, marked as answer]
First the table is created 

https://github.com/pgpartman/pg_partman/blob/master/sql/functions/create_partition_time.sql#L170

then it is attached

https://github.com/pgpartman/pg_partman/blob/master/sql/functions/create_partition_time.sql#L214

Same is true for integer based partitioning as well

**Reply by jhchee** [reply]
@keithf4, thanks for your prompt response, this solves my doubt!
